rust/phoenix-sdk/src/sdk_client.rs

```diff
@@ -522,4 +524,118 @@ impl SDKClient {
         let cancels = self.parse_cancels(&signature).await;
         Some((signature, cancels))
     }
+
+    // Useful if unsure trader has an ATA for the base or quote or has a seat on the market
+    pub async fn get_limit_order_new_maker_ixs(
+        &self,
+        price: u64,
+        side: Side,
+        size: u64,
+    ) -> Vec<Instruction> {
+        let base_ata_instruction_option =
+            create_ata_ix_if_needed(&self.client, &self.trader, &self.trader, &self.base_mint)
+                .await;
+
+        let quote_ata_instruction_option =
+            create_ata_ix_if_needed(&self.client, &self.trader, &self.trader, &self.quote_mint)
+                .await;
+
+        let claim_seat_ix_option =
+            create_claim_seat_ix_if_needed(&self.client, &self.active_market_key, &self.trader)
+                .await;
+
+        let limit_order_ix = self.get_limit_order_ix(price, side, size);
+
+        let mut instructions = Vec::with_capacity(4);
+        if let Some(base_ata_instruction) = base_ata_instruction_option {
+            instructions.push(base_ata_instruction);
+        }
+        if let Some(quote_ata_instruction) = quote_ata_instruction_option {
+            instructions.push(quote_ata_instruction);
+        }
+        if let Some(claim_seat_ix) = claim_seat_ix_option {
+            instructions.push(claim_seat_ix);
+        }
+        instructions.push(limit_order_ix);
+        instructions
+    }
+
+    // Useful if known that trader has an ATA for the base or quote through prior interactions but may have been evicted
+    pub async fn get_limit_order_existing_maker_ixs(
+        &self,
+        price: u64,
+        side: Side,
+        size: u64,
+    ) -> Vec<Instruction> {
+        let claim_seat_ix_option =
+            create_claim_seat_ix_if_needed(&self.client, &self.active_market_key, &self.trader)
+                .await;
+
+        let limit_order_ix = self.get_limit_order_ix(price, side, size);
+
+        let mut instructions = Vec::with_capacity(2);
+
+        if let Some(claim_seat_ix) = claim_seat_ix_option {
+            instructions.push(claim_seat_ix);
+        }
+        instructions.push(limit_order_ix);
+        instructions
+    }
+
+    // Useful if unsure trader has an ATA for the base or quote or has a seat on the market
+    pub async fn get_post_only_new_maker_ixs(
+        &self,
+        price: u64,
+        side: Side,
+        size: u64,
+    ) -> Vec<Instruction> {
+        let base_ata_instruction_option =
+            create_ata_ix_if_needed(&self.client, &self.trader, &self.trader, &self.base_mint)
+                .await;
+
+        let quote_ata_instruction_option =
+            create_ata_ix_if_needed(&self.client, &self.trader, &self.trader, &self.quote_mint)
+                .await;
+
+        let claim_seat_ix_option =
+            create_claim_seat_ix_if_needed(&self.client, &self.active_market_key, &self.trader)
+                .await;
```

Comment:
If we wanted to optimize this, we could potentially do 1 network call with `getMultipleAccounts` and then compute based off of this state. Not important though